src/app/epics/user-login.epic.spec.ts

```diff
@@ -0,0 +1,176 @@
+                .subscribe(
+                    () => {},
+
+                    () => done.fail('User Login Epic: login fail'),
+
+                    () => { 
+                        expect( mockLoginService.login )
+                            .toHaveBeenCalledWith(user, pass);
+                        expect( mockUserActions.loginSuccess )
+                            .toHaveBeenCalledWith(user, date);
+                        expect( mockUserActions.loginFailed )
+                            .not.toHaveBeenCalled();
+                        done();
+                    }
+                );
+        });
+        it('should throw error and failed action when login fails', (done: DoneFn) => {
+            const user: string = 'cheese';
+            const pass: string = 'cheese';
+            const date: number = Date.now();
+            const mockErrorMsg: string = 'MOCK_ERROR';
+
+            spyOn(mockLoginService, 'login')
+                .and
+                .returnValue( Observable.throw(mockErrorMsg) );
+            spyOn(mockUserActions, 'loginSuccess')
+                .and
+                .returnValue( mockUserActions.loginSuccess(user, date) );
+            spyOn(mockUserActions, 'loginFailed')
+                .and
+                .returnValue( mockUserActions.loginFailed(mockErrorMsg) );
+
+            action$ = ActionsObservable.of({ 
+                type: UserActions.LOGIN_USER_REQUEST,
+                username: user,
+                pass: pass,
+            });
+
+            userLoginEpic.login(action$)
+            .subscribe(
+                () => {},
+
+                () => done.fail('User Login Epic: login fail'),
+
+                () => { 
+                    expect( mockLoginService.login )
+                        .toHaveBeenCalledWith(user, pass);
+                    expect( mockUserActions.loginFailed )
+                        .toHaveBeenCalledWith(mockErrorMsg);
+                    expect( mockUserActions.loginSuccess )
+                        .not.toHaveBeenCalled();
+                    done();
+                }
+            );
+
+        
+        });
+
+    });
+
+    describe('logout', () => {
+        beforeEach(() => {
+            mockLoginService = new MockLoginService();
+            mockUserActions = new MockUserActions();
+            mockRouter = new MockRouter();
+            userLoginEpic = new UserLoginEpic(
+                mockLoginService as any, 
+                mockUserActions as any,
+                mockRouter as any
+            );
+        });
+        it('should initiate logout and emit action', (done: DoneFn) => {
+            spyOn(mockLoginService, 'logout');
+
+            action$ = ActionsObservable.of({ 
+                type: UserActions.LOGOUT_USER,
+            });
+
+            userLoginEpic.logout(action$)
+                .subscribe(
+                    (val) => {
+                        expect( val ).not.toBeDefined();
```

Comment:
I don't know if there's much value in asserting that no other actions were emitted - certainly asserting that logout was called is definitely valid, also keeps with single expectations per test

You _could_ assert that no other actions should emit as a way to confirm that no other state in your store should be modified by this epic, if that were the case, then it could be a separate test case